src/main/java/calculator/Main.java

```diff
@@ -0,0 +1,142 @@
+package calculator;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Scanner;
+
+public class Main {
+    public static HashMap<Character, Operation> operations = new HashMap<>();
```

Comment:
Reference must be the interface, the object -> the implementation.
Thus: public static Map<Character, Operation> operations = new HashMap<>();